### PR TITLE
miru: init at 3.11.6

### PR DESCRIPTION
**Pull request by LyonSyonII**
- miru: init at 3.11.6

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix
Comment (sikmir) [inline maintainers/maintainer-list.nix:9665]: Please, add yourself to maintainer-list.nix as separate commit.

Comment (LyonSyonII) [reply maintainers/maintainer-list.nix:9665]: Sorry, do you mean a different PR?

I've renamed the commit if this is what you mean.

Comment (sikmir) [reply maintainers/maintainer-list.nix:9665]: No, just separate commit within this PR. And squash current 5 commits into single one.

Comment (LyonSyonII) [reply maintainers/maintainer-list.nix:9665]: Done

File: pkgs/applications/video/miru/default.nix
Comment (Janik-Haag) [inline pkgs/applications/video/miru/default.nix:38]: Does it not work on macos or is it supposed to only work on linux?

**Review by Janik-Haag — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/miru/default.nix
Comment (LyonSyonII) [reply pkgs/applications/video/miru/default.nix:38]: It's an AppImage, I don't know if those work in macos.

Comment (AndersonTorres) [reply pkgs/applications/video/miru/default.nix:38]: > Does it not work on macos or is it supposed to only work on linux?

Appimages are linux-only. MacOS uses other format.

Comment (Janik-Haag) [reply pkgs/applications/video/miru/default.nix:38]: then please change it to `platforms = platforms.linux;`

Comment (AndersonTorres) [reply pkgs/applications/video/miru/default.nix:38]: I am not so sure, because there is more than x86_64 on (our?) Linux.

The elegant way is to intersect `linux` with `x86_64`.

File: pkgs/applications/video/miru/default.nix
Comment (AndersonTorres) [reply pkgs/applications/video/miru/default.nix:18]: `hash` does not accept this blank type.
```suggestion
    hash = "<convert to sri>";
```

Run `nix hash to-sri sha256:0abc03c5a8cf9557309d054d2281b4267951f5d929c8ee11d4f015ed335ef2f2`

File: pkgs/applications/video/miru/default.nix
Comment (Janik-Haag) [inline pkgs/applications/video/miru/default.nix:37]:
```suggestion
  description = "Bittorrent streaming software for cats. Stream anime torrents, real-time with no waiting for downloads (If xdg-open isn't working on login, try using ungoogled-chromium)";
  homepage = "https://github.com/ThaUnknown/miru/";
  license = lib.licenses.gpl3Only;
```

File: pkgs/applications/video/miru/default.nix
Comment (Janik-Haag) [inline pkgs/applications/video/miru/default.nix:10]:
```suggestion
```

File: pkgs/applications/video/miru/default.nix
Comment (Janik-Haag) [inline pkgs/applications/video/miru/default.nix:17]:
```suggestion
    url = "${homepage}/releases/download/v${version}/linux-Miru-${version}.AppImage";
```

File: pkgs/applications/video/miru/default.nix
Comment (Janik-Haag) [reply pkgs/applications/video/miru/default.nix:38]: Ah I'm sorry it off-course only works on x86_64-linux.

**Review by Janik-Haag — COMMENTED** [review]
Mostly good now.

File: pkgs/applications/video/miru/default.nix
Comment (Janik-Haag) [inline pkgs/applications/video/miru/default.nix:9]:
```suggestion
    hash = "sha256-CrwDxajPlVcwnQVNIoG0JnlR9dkpyO4R1PAV7TNe8vI=";
```

I don't know where you got your hash from but when I'm downloading it I get the one above. You also need to specify the hash type with `sha256-`

**Review by Janik-Haag — COMMENTED** [review]
It should build fine after applying these patches and should be good to go. Sorry that it took so long. You can test if it works fine locally by building it with `nix-build -A miru` in the root of the local nixpkgs, You can then find the build output in `./results/`

**Comment by LyonSyonII** [comment]
> It should build fine after applying these patches and should be good to go. Sorry that it took so long. You can test if it works fine locally by building it with `nix-build -A miru` in the root of the local nixpkgs, You can then find the build output in `./results/`

Thank you for the changes! I'll try building it.

**Review by Janik-Haag — COMMENTED** [review]
At last you need to sqaush the commits so you only have `miru: init at 3.11.6` and `maintainers: add lyonsyonii`it should be good to go after that.

File: pkgs/applications/video/miru/default.nix
Comment (AndersonTorres) [inline pkgs/applications/video/miru/default.nix:2]:
```suggestion
{ lib
, appimageTools
, fetchurl
}:

appimageTools.wrapType2 rec {
```

File: pkgs/applications/video/miru/default.nix
Comment (AndersonTorres) [inline pkgs/applications/video/miru/default.nix:5]:
```suggestion
  pname = "miru";
  version = "3.11.6";
  name = "miru";
```

File: pkgs/applications/video/miru/default.nix
Comment (AndersonTorres) [inline pkgs/applications/video/miru/default.nix:8]:
```suggestion
    url = "https://github.com/ThaUnknown/miru/releases/download/v${version}/linux-Miru-${version}.AppImage";
```

File: pkgs/applications/video/miru/default.nix
Comment (AndersonTorres) [inline pkgs/applications/video/miru/default.nix:23]:
```suggestion
  extraInstallCommands = let
    contents = appimageTools.extractType2 { name = "linux-Miru-${version}.AppImage"; inherit src; };
  in ''
    mkdir -p "$out/share/applications"
    mkdir -p "$out/share/lib/miru"
    cp -r ${contents}/{locales,resources} "$out/share/lib/miru"
    cp -r ${contents}/usr/* "$out"
    cp "${contents}/${pname}.desktop" "$out/share/applications/"
    substituteInPlace $out/share/applications/${pname}.desktop --replace 'Exec=AppRun' 'Exec=${pname}'
  '';
```

File: pkgs/applications/video/miru/default.nix
Comment (AndersonTorres) [inline pkgs/applications/video/miru/default.nix:26]:
```suggestion
    description = "Bittorrent streaming software for cats";
```

File: pkgs/applications/video/miru/default.nix
Comment (AndersonTorres) [inline pkgs/applications/video/miru/default.nix:30]: Set `meta.mainProgram` and `meta.sourceProvenance` too.

File: pkgs/top-level/all-packages.nix
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:10236]: Is this a networking app, a video app, or both?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by donovanglover** [comment]
Closing since https://github.com/NixOS/nixpkgs/pull/172872 is in nixpkgs now. Thanks for your contribution!

**Pull request closed without merging**
